### ISSUE TITLE
Kura Wires severity level based filtering issue

### ISSUE THREAD

**Issue by nicolatimeus** [eclipse/kura#1056]
`WireSupportImpl.filter()` returns a new WireRecord for every WireField it founds:
```java
/** {@inheritDoc} */
    @Override
    public List<WireRecord> filter(final List<WireRecord> records) {
        requireNonNull(records, s_message.wireRecordsNonNull());
        final SeverityLevel level = this.getSeverityLevel();
        // If the severity level is SEVERE, then all wire fields remain
        if ((level == null) || (level == SEVERE)) {
            return records;
        }
        final List<WireRecord> newRecords = CollectionUtil.newArrayList();
        final List<WireField> newFields = CollectionUtil.newArrayList();
        for (final WireRecord wireRecord : records) {
            ==> for (final WireField wireField : wireRecord.getFields()) {
                // If the severity level is INFO, then only info wire fields
                // will remain
                final SeverityLevel wireFieldLevel = wireField.getSeverityLevel();
                if ((wireFieldLevel == INFO) && (level == INFO)) {
                    newFields.add(wireField);
                }
                // If the severity level is ERROR, then only ERROR wire fields
                // will remain
                if ((wireFieldLevel == ERROR) && (level == ERROR)) {
                    newFields.add(wireField);
                }
                // If the severity level is CONFIG, then info and CONFIG wire
                // fields remain
                if (((wireFieldLevel == INFO) || (wireFieldLevel == CONFIG)) && (level == CONFIG)) {
                    newFields.add(wireField);
                }
                ==> final WireRecord newWireRecord = new WireRecord(wireRecord.getTimestamp(), wireRecord.getPosition(),
                        newFields);
                ==> newRecords.add(newWireRecord);
            }
        }
        return newRecords;
    }
```
The spurious results of the filtering can be seen by setting the severity level of any Logger component to something different than SEVERE.

**Comment by amitjoy** [comment]
Good catch. It's a misplacement of parenthesis. I will quick fix it tomorrow.

**Comment by amitjoy** [comment]
As `WireSupporImpl` would be moved to new bundle, I would fix it as soon as #1047 gets merged.
